### PR TITLE
Migrate prophet.hdays countries

### PR DESCRIPTION
**Pull request by arkid15r**
This PR starts a series of changes in order to resolve "help-wanted" labeled https://github.com/facebook/prophet/issues/2011:
  - Add  Kyrgyzstan, Philippines, Thailand initial support.
  - Update Belarus, Georgia, India, Indonesia, Russia.
  - Refactor tests.

### CODE REVIEW

File: holidays/countries/india.py [hunk 1]
```diff
@@ -250,6 +254,37 @@ def _populate(self, year):
             hol_date = date(year, *holi_dates[year])
             self[hol_date] = "Holi"
 
+        # Islamic holidays
+        # Day of Ashura (10th day of 1st Islamic month)
+        name = "Day of Ashura"
+        for dt in _islamic_to_gre(year, 10, 1):
```
Comment (KJhellico) [inline holidays/countries/india.py:260]:
```suggestion
        for dt in _islamic_to_gre(year, 1, 10):
```

File: holidays/countries/india.py [hunk 2]
```diff
@@ -250,6 +254,37 @@ def _populate(self, year):
...
+            self[dt] = f"{name}* (*estimated)"
+
+        # Mawlid, Birth of the Prophet (12th day of 3rd Islamic month)
+        name = "Mawlid"
+        for dt in _islamic_to_gre(year, 10, 1):
```
Comment (KJhellico) [inline holidays/countries/india.py:265]:
```suggestion
        for dt in _islamic_to_gre(year, 3, 12):
```

File: holidays/countries/philippines.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+from holidays.holiday_base import HolidayBase
+from holidays.utils import _ChineseLuniSolar, _islamic_to_gre
+
+
+class Philippines(HolidayBase):
+    """
+    Philippines holidays.
+
+    References:
+      - https://en.wikipedia.org/wiki/Public_holidays_in_Thailand
```
Comment (KJhellico) [inline holidays/countries/philippines.py:27]:
```suggestion
      - https://en.wikipedia.org/wiki/Public_holidays_in_the_Philippines
```

File: holidays/countries/philippines.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+#
+#  Authors: dr-prodigy <maurizio.montel@gmail.com> (c) 2017-2022
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date
+
+from dateutil.easter import easter
+from dateutil.relativedelta import relativedelta as rd
```
Comment (KJhellico) [inline holidays/countries/philippines.py:15]:
```suggestion
from dateutil.relativedelta import MO
from dateutil.relativedelta import relativedelta as rd
```

File: holidays/countries/philippines.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        # Eid al-Adha.
+        for dt in _islamic_to_gre(year, 12, 10):
+            self[dt] = "Eid'l Adha"
+
+        # Ninoy Aquino Day.
+        self[date(year, AUG, 21)] = "Ninoy Aquino Day"
+
+        # National Heroes Day.
+        self[date(year, AUG, 27)] = "National Heroes Day"
```
Comment (KJhellico) [inline holidays/countries/philippines.py:77]:
```suggestion
        self[date(year, AUG, 31) + rd(weekday=MO(-1))] = "National Heroes Day"
```

File: holidays/countries/thailand.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+        # Labour day.
+        add_holiday(date(year, MAY, 1), "Labour Day")
+
+        # Buddha's Birthday.
+        add_holiday(self.cnls.vesak_may_date(year), "Buddha's Birthday")
+
+        # Coronation Day.
+        if year <= 2016:
+            add_holiday(date(year, MAY, 4), "Coronation Day")
```
Comment (KJhellico) [inline holidays/countries/thailand.py:85]:
```suggestion
        if year <= 2016:
            add_holiday(date(year, MAY, 5), "Coronation Day")
        elif year >= 2020:
            add_holiday(date(year, MAY, 4), "Coronation Day")
```

**Review by KJhellico — CHANGES_REQUESTED** [review]

**Review by arkid15r — COMMENTED** [review]
I've accepted all the suggestions and fixed existing tests.

Thanks again!

**Pull request merged**